lua/nvim-highlight-colors/buffer_utils.lua

```diff
@@ -26,10 +26,10 @@ function M.get_positions_by_regex(patterns, min_row, max_row, row_offset)
 				local column_offset = last_repeated_color and last_repeated_color.end_column or nil
 
 				local pattern_without_usage_regex = M.remove_color_usage_pattern(match)
-				local start_column = vim.fn.match(value, pattern_without_usage_regex, column_offset)
-				local end_column = vim.fn.matchend(value, pattern_without_usage_regex, column_offset)
+				local t = vim.fn.matchstrpos(value, pattern_without_usage_regex, column_offset)
+				local start_column, end_column = t[2], t[3]
 
-				if (row >= 0) then
+				if (row >= -1) then
```

Comment:
This one is giving an error when the plugin is used with `first_column`, i already tried fixing this with this render type, but i guess it's not possible, i would recommend receiving `render` as param in this fn and change the value according to it
```
local min_row = render == 'first_column' and 0 or -1
if (row >= min_row)
```

Reply:
Also please keep start_column and end_column as is, no need to use `matchstrpos` in here

Reply:
> This one is giving an error when the plugin is used with `first_column`, i already tried fixing this with this render type, but i guess it's not possible, i would recommend receiving `render` as param in this fn and change the value according to it
> 
> ```
> local min_row = render == 'first_column' and 0 or -1
> if (row >= min_row)
> ```

I did check, no errors here because of the code in init.lua

And what do you mean no need to use matchstrpos ? It just makes sense to use single function than two ? Do explain rather than the vague comments. Yes i stress tested it also, matchstrpos is faster too.

Reply:
Here is a screen recording. I don't know why it isn't working for you, send some logs.

https://user-images.githubusercontent.com/32216346/187029382-932ba170-103d-441a-bef6-cfedd7180d24.mp4

Reply:
The code works, but it's not right, `get_positions_by_regex` should do exactly what is says, return what should be highlighted. There should be no further logic to detect wether a highlight should occur or not, if we follow this path `single source of truth`, it's easier to maintain the code.
So please remove that if statement from `init.lua` and follow the logic i've added on the comment above.

Reply:
Well by that logic, by passing the render options to  get_positions_by_regex function, you are contradicting your own logic that "There should be no further logic to detect whether a highlight should occur or not".

Anyway, being said that. I will drop that commit because it anyway gonna get fixed when i remove the window logic ( i still have no idea why it exists given it can just be done by normal highlight of extmarks, or, maybe i am wrong ).

Reply:
Please don't need to worry with this project anymore, i am already spending too much time discussing this with you.

Reply:
Please feel free to fork the project and create your own iteration of it, with your own point of view and code style

Reply:
Fair enough, thank you for the co-operation though.

Your idea of just applying the highlights to the visible content is awesome, that's why i was attracted to this plugin and wanted improve this.

Anyway, waiting for the bug fixes, xD.